### PR TITLE
[BugFix][Ops][310p]:fix the accuracy issue caused by MoEGatingTopkSoftmax

### PR DESCRIPTION
**Pull request by Tflowers-0129**
### What this PR does / why we need it?

Based on previous community issues, we had already noticed that Qwen3.5-MoE might have a hidden accuracy issue. The root cause has now been identified: when the input token dimension of the `MoEGatingTopkSoftmax` operator is 2048, UB overflow may occur, which can eventually produce INF values and lead to accuracy degradation.

Since the operator-level fix may not be externally available, this PR first introduces a workaround at the vllm-ascend level.

### Does this PR introduce _any_ user-facing change?

NA

### How was this patch tested?

Local test

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/b9a7cd464c9ae9b1b450f8982b76d7be4de73724

### CODE REVIEW

**Comment by Tflowers-0129** [comment]
The accuracy results and performance impact are shown below:
Gsm8k accuracy：
image


TextVqa accuracy：
image



the Performance:
***The performance impact is manageable. It mainly affects scenarios with Prefill requests, with an estimated regression of about 150–200 us per layer. Overall, the impact on performance is relatively small.***

image

**Comment by Tflowers-0129** [comment]
算子精度问题测试脚本：
```
import torch
import torch_npu

torch.manual_seed(0)
torch.npu.set_compile_mode(jit_compile=False)

# 2048 个 token 使用完全相同的 router logits，正确结果也应该完全相同。
router_logits = torch.randn(1, 256, dtype=torch.float16).repeat(2048, 1).npu()

# 原始调用：2048 个 token 一次送入算子。
full_weights, full_ids, _ = torch_npu.npu_moe_gating_top_k_softmax(
    router_logits, k=8
)

# 对照调用：相同输入按 1024 个 token 分块。
part1 = torch_npu.npu_moe_gating_top_k_softmax(router_logits[:1024], k=8)
part2 = torch_npu.npu_moe_gating_top_k_softmax(router_logits[1024:], k=8)
chunk_weights = torch.cat([part1[0], part2[0]], dim=0)
chunk_ids = torch.cat([part1[1], part2[1]], dim=0)

torch.npu.synchronize()

print("full batch 非有限权重数量:", (~torch.isfinite(full_weights)).sum().item())
print("chunked 非有限权重数量:", (~torch.isfinite(chunk_weights)).sum().item())
print("full batch 与 chunked 的 expert id 差异数量:", (full_ids != chunk_ids).sum().item())
print("full batch 与 chunked 的最大权重差:", (full_weights - chunk_weights).abs().max().item())
```
表现：
```
full batch 非有限权重数量: 1792
chunked 非有限权重数量: 0
full batch 与 chunked 的 expert id 差异数量: 16384
full batch 与 chunked 的最大权重差: 65504.0
```

**Comment by Tflowers-0129** [comment]
Fix this issue: https://github.com/vllm-project/vllm-ascend/issues/10892

**Review by MengqingCao — APPROVED** [review]
